exporters/otlp/internal/transform/metric.go

```diff
@@ -130,10 +130,18 @@ func minMaxSumCount(desc *metricsdk.Descriptor, labels metricsdk.Labels, a aggre
 	}, nil
 }
 
-// stringKeyValues transforms a KeyValues into an OTLP StringKeyValues.
-func stringKeyValues(kvs []core.KeyValue) []*commonpb.StringKeyValue {
-	result := make([]*commonpb.StringKeyValue, 0, len(kvs))
-	for _, kv := range kvs {
+// stringKeyValues transforms a label iterator into an OTLP StringKeyValues.
+func stringKeyValues(iter metricsdk.LabelIterator) []*commonpb.StringKeyValue {
+	l := iter.Len()
+	if l == 0 {
+		// TODO: That looks like a pointless allocation in case of
```

Comment:
I believe the compiler is clever enough not to actually allocate here. I imagine we could fix the test, but a non-nil empty slice is OK with me. (Leaving the TODO is good though.)